### PR TITLE
test(common): update all tests according to the common approach

### PR DESCRIPTION
**Pull request by kulenkov**
# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated tests.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: cypress/integration/landing_page_spec.ts [hunk 1]
```diff
@@ -7,24 +7,20 @@ describe('Landing Page test suite', () => {
...
-        .should('be.visible');
-      cy.get(landing.sloganBs)
-        .should('be.visible');
-      cy.get(landing.descriptionBs)
-        .should('be.visible');
-      cy.get(landing.versionBs)
-        .should('be.visible');
-      cy.get(landing.advantagesBs)
-        .should('be.visible');
+      landing.isElementVisible(landing.mainClass, landing.logoAtContent);
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts:17]: move this checks into previous 'it'. If you don't make any actions, and make only assertions - better to serve them in 1 'it'

File: cypress/support/landing.po.ts [hunk 2]
```diff
@@ -21,4 +30,37 @@ export class LandingPo extends BaseComponent {
...
+    cy.get('body').should('have.class', this.mobileMenuOpened);
+  }
+
+  isMenuHasDescendants(descen1Selector: string, descen2Selector: string) {
+    cy.get(this.mobileMenu).should('be.visible')
+      .and('have.descendants', descen1Selector)
+      .and('have.descendants', descen2Selector);
+  }
+
+  loadBootstrapVersion(bsVersion: number) {
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: visitNGXBootstrapWithBsVersion

File: cypress/integration/landing_page_spec.ts [hunk 3]
```diff
@@ -73,8 +63,45 @@ describe('Landing Page test suite', () => {
...
+    it('Main-menu in mobile view', () => {
+      cy.viewport(375, 667);
+      landing.clickOnMenuBtn();
+      landing.isMobileMenuActive();
+      landing.isMenuHasDescendants(bootstrapVer, demosList);
+    });
+  });
+
+  describe('Load Bootstrap version', () => {
+    const bootStrapVer3 = '3';
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts]: remove

File: cypress/integration/landing_page_spec.ts [hunk 4]
```diff
@@ -73,8 +63,45 @@ describe('Landing Page test suite', () => {
...
+    const bootStrapVer4 = '4';
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts]: remove

File: cypress/integration/landing_page_spec.ts [hunk 5]
```diff
@@ -73,8 +63,45 @@ describe('Landing Page test suite', () => {
...
+
+    it('the 3rd version is ', () => {
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts:91]: const bsVersion = 3;

File: cypress/integration/landing_page_spec.ts [hunk 6]
```diff
@@ -73,8 +63,45 @@ describe('Landing Page test suite', () => {
...
+      landing.loadBootstrapVersion(3);
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts]: use bsVersion instead of number

File: cypress/integration/landing_page_spec.ts [hunk 7]
```diff
@@ -73,8 +63,45 @@ describe('Landing Page test suite', () => {
...
+      landing.isBootstrapVersion(bootStrapVer3);
+    });
+
+    it('the 4th version is ', () => {
```
Comment (ludmilanesvitiy) [inline cypress/integration/landing_page_spec.ts:97]: const bsVersion = 4;

File: cypress/support/base.component.ts [hunk 8]
```diff
@@ -192,4 +192,18 @@ export abstract class BaseComponent {
       .invoke('text')
       .should('to.contains', expectedTxt);
   }
+
+  isElementVisible(baseSelector: string, elementToFind: string, elemNumber = 0) { // TODO remove after the 'e2e-typeahead' will be merged
+    cy.get(`${baseSelector} ${elementToFind}`).eq(elemNumber).should('be.visible');
+  }
+
+  isUrlExist(urlToCheck: string) {
```
Comment (ludmilanesvitiy) [inline cypress/support/base.component.ts]: urlToCheck => expectedUrl

File: cypress/support/base.component.ts [hunk 9]
```diff
@@ -192,4 +192,18 @@ export abstract class BaseComponent {
...
+    cy.url().should('include', urlToCheck);
+  }
+
+  isElemHasCorrectUrl(elementSelector: string, link: string) {
```
Comment (ludmilanesvitiy) [inline cypress/support/base.component.ts]: link => expectedUrl

File: cypress/support/base.component.ts [hunk 10]
```diff
@@ -192,4 +192,18 @@ export abstract class BaseComponent {
...
+    cy.get(`${elementSelector} [href="${ link }"]`)
+      .should('have.attr', 'href')
+      .and('equal', link);
```
Comment (ludmilanesvitiy) [inline cypress/support/base.component.ts]: to.equal

File: cypress/support/landing.po.ts [hunk 11]
```diff
@@ -2,15 +2,24 @@ import { BaseComponent } from './base.component';
 
 export class LandingPo extends BaseComponent {
   pageUrl = '/';
+  documentationUrl = '/documentation';
 
+  mainClass = '.main';
   logoAtHeader = '.logo';
   logoAtContent = '.content-logo';
+  headerSelector = '.header';
+  stackOverBtn = 'li:nth-child(1) a';
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: a[href*= "stackoverflow"]

File: cypress/support/landing.po.ts [hunk 12]
```diff
@@ -2,15 +2,24 @@ import { BaseComponent } from './base.component';
...
+  gitHbBtn = 'li:nth-child(2) a';
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: .header-list a[href*= "github"]

File: cypress/support/landing.po.ts [hunk 13]
```diff
@@ -2,15 +2,24 @@ import { BaseComponent } from './base.component';
...
+  slackNgxBtn = 'li:nth-child(3) a';
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: .header-list a[href*= "slack"]

File: cypress/support/landing.po.ts [hunk 14]
```diff
@@ -21,4 +30,37 @@ export class LandingPo extends BaseComponent {
   mitLicenseUrl = 'https://github.com/valor-software/ngx-bootstrap/blob/development/LICENSE';
   crCommonsUrl = 'https://creativecommons.org/licenses/by/3.0/';
   originalBsUrl = 'https://getbootstrap.com/';
+
+  isNavigateBtnExist(btnName: string, elementIndex: number) {
+    cy.get(this.navBtn).contains(btnName)
+      .should('be.enabled');
+  }
+
+  clickOnMenuBtn() {
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: clickOnMobileMenuBtn

File: cypress/support/landing.po.ts [hunk 15]
```diff
@@ -21,4 +30,37 @@ export class LandingPo extends BaseComponent {
...
+    cy.get('.sidebar-list').eq(1)
+      .should('have.descendants', 'li').invoke('text')
+      .should('contain', searchResult);
+  }
+
+  isMobileMenuActive() {
+    cy.get('body').should('have.class', this.mobileMenuOpened);
+  }
+
+  isMenuHasDescendants(descen1Selector: string, descen2Selector: string) {
```
Comment (ludmilanesvitiy) [inline cypress/support/landing.po.ts]: isMobileMenuHasDescendants

**Review by ludmilanesvitiy — CHANGES_REQUESTED** [review]

**Review by ludmilanesvitiy — DISMISSED** [review]

**Comment by Domainv** [comment]
Needs to resolve confilcts

**Review by ludmilanesvitiy — APPROVED** [review]

**Pull request merged**